### PR TITLE
ipvlan: add IPVLAN interface support

### PR DESCRIPTION
**Pull request by ffmancera**
Example:

```
---
interfaces:
  - name: ipvlan0
    type: ipvlan
    state: up
    ipvlan:
      base-iface: eth1
      mode: l2
```

Integration tests added.

Resolves: https://issues.redhat.com/browse/RHEL-56571

### CODE REVIEW

File: rust/src/lib/ifaces/ipvlan.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+#[non_exhaustive]
+pub struct IpVlanConfig {
+    pub base_iface: String,
+    pub mode: IpVlanMode,
```
Comment (cathay4t) [inline rust/src/lib/ifaces/ipvlan.rs]: Both base-iface and mode can be optional when modifying existing interface. It also helps us on `gen_diff()`.

Comment (cathay4t) [reply rust/src/lib/ifaces/ipvlan.rs]: Please also mark `mode` with `derive[serde(default)]`, user does not need to define it for default value.

**Review by cathay4t — CHANGES_REQUESTED** [review]

File: rust/src/lib/ifaces/ipvlan.rs (hunk 1 above)
Comment (ffmancera) [reply rust/src/lib/ifaces/ipvlan.rs]: Sure! Let me adjust it, thanks!

Comment (ffmancera) [reply rust/src/lib/ifaces/ipvlan.rs]: Done, thank you!

**Review by cathay4t — DISMISSED** [review]
Looks good except we do not validate whether `base-iface` exists.

**Review by cathay4t — APPROVED** [review]

**Pull request merged**